### ISSUE TITLE
lsp: give warning/error message if automatic caching 404's

### ISSUE THREAD

**Issue by grian32** [denoland/deno#11049]
If you `right-click -> cache "link"` and the link leads to a 404 then it does nothing, had this happen where i accidentally left in a `'` at the end of a link after copy pasting.

**Comment by kitsonk** [comment]
What would you expect to happen?

**Comment by grian32** [comment]
> What would you expect to happen?

An error or warning saying the link is invalid, i think in vscode this would equal to one of those popups on the bottom right.

**Comment by kitsonk** [comment]
It seems like a reasonable feature, though we would need to consider how to make it non-spammy, as currently we don't know if a missing dependency is because of a request to cache or because you opened up the file, and if you got a notification every time you opened up a file it would get very annoying.

The other option would be to expose a more informative diagnostic so that an uncached module is different then one we have attempted to cached but got a 404. The problem with that is that we don't hold onto the response, and if we did hold onto the response, how do we invalidate it, as the module may appear in the future.
